mmv1/products/datapipeline/api.yaml

```diff
@@ -0,0 +1,398 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: DataPipeline
+display_name: Dataflow Pipeline
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://datapipelines.googleapis.com/v1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Data Pipelines API
+    url: https://console.cloud.google.com/apis/library/datapipelines.googleapis.com
+objects:
+  - !ruby/object:Api::Resource
+    name: Pipeline
+    base_url: projects/{{project}}/locations/{{region}}/pipelines
+    update_verb: :PATCH
+    description: |
+      The main pipeline entity and all the necessary metadata for launching and managing linked jobs.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/dataflow'
+      api: 'https://cloud.google.com/dataflow/docs/reference/data-pipelines/rest/v1/projects.locations.pipelines'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'region'
+        description: 'A reference to the region'
+        input: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          "The pipeline name. For example': 'projects/PROJECT_ID/locations/LOCATION_ID/pipelines/PIPELINE_ID."
+          "- PROJECT_ID can contain letters ([A-Za-z]), numbers ([0-9]), hyphens (-), colons (:), and periods (.). For more information, see Identifying projects."
+          "LOCATION_ID is the canonical ID for the pipeline's location. The list of available locations can be obtained by calling google.cloud.location.Locations.ListLocations. Note that the Data Pipelines service is not available in all regions. It depends on Cloud Scheduler, an App Engine application, so it's only available in App Engine regions."
+          "PIPELINE_ID is the ID of the pipeline. Must be unique for the selected project and location."
+        input: true
+        required: true
```

Comment:
Check out the follow patterns we have for name.
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/cloudfunctions2/Function.yaml#L213-L222

I would recommend using a custom flattener and expansion

```
    custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.erb'
    custom_expand: 'templates/terraform/custom_expand/shortname_to_url.go.erb'
```

and specifying the short name url as 
```
id_format: 'projects/{{project}}/locations/{{location}}/functions/{{name}}'
import_format:
  ['projects/{{project}}/locations/{{location}}/functions/{{name}}']
````